engine/apps/telegram/tasks.py

```diff
@@ -101,6 +102,16 @@ def send_link_to_channel_message_or_fallback_to_full_alert_group(
         else:
             # seems like the message won't appear in Telegram channel, so send the full alert group to user
             user_connector.send_full_alert_group(alert_group=alert_group, notification_policy=notification_policy)
+    except AlertGroupTelegramMessageDoesNotExist as e:
+        # On the first run, if alert group message doesn't exist in db, restart the task without retry.
+        # The message is expected to appear by the next call, otherwise Exception will be raised.
+        if is_first_message_check:
+            logger.warning(f"{e}. Restarting task")
+            send_link_to_channel_message_or_fallback_to_full_alert_group.apply_async(
+                (alert_group_pk, notification_policy_pk, user_connector_pk, False), countdown=3
```

Comment:
Curious, did you consider delaying the original `send_link_to_channel_message_or_fallback_to_full_alert_group.delay` call in `telegram/models/connectors/personal.py` instead? Wondering if we just set the `countdown` there would be enough and we don't need to add the extra complexity here (which seems to be a custom/manual retry without the exception :-)). Also, how often do we get this error? (thinking if this failing may indicate some other issue causing the delayed message creation when happening)

Reply:
It happens not often. The case is when the first escalation step is telegram notification, sometimes we don't have alert group telegram message saved in our db by this moment. The issue resolves after one retry.
Your proposal to add a delay to call in `telegram/models/connectors/personal.py` makes sense, I was just thinking about making this more visible, because otherwise if we will call this function from other places in the future we will need to always call this task with a delay, that is only needed to cover this rare case

Reply:
Decided to add a delay to the initial call in `telegram/models/connectors/personal.py` since in this case recalling the task to avoid retry seems to be overcomplication 👍